Add Azure Identity CCO/OLM docs

adellape opened the pull request.

adellape commented:
/retest

adellape commented:
/retest

adellape commented on _attributes/common-attributes.adoc:
```diff
@@ -240,6 +240,7 @@ endif::[]
 //Azure
 :azure-full: Microsoft Azure
 :azure-short: Azure
+:azure-id: Azure AD Workload Identity
```
Checking with Jeana when they're back re: the naming here.

adellape commented on modules/olm-installing-from-operatorhub-using-web-console.adoc:
```diff
@@ -77,18 +77,25 @@ endif::[]
 image::oadp-install-operator-role-arn.png[Entering the ARN]
 To create the role's ARN, follow the procedure described in link:https://access.redhat.com/documentation/en-us/red_hat_openshift_service_on_aws/4/html/tutorials/cloud-experts-deploy-api-data-protection#prepare-aws-account_cloud-experts-deploy-api-data-protection[Preparing AWS account].
 
+* If the cluster is in {azure-id} mode, add the client ID, tenant ID, and subscription ID in the appropriate field.
```
When I originally added the AWS STS bullet before this Azure one, I did not include a screenshot, tho it looks like a different PR has since added one in. I don't really want to include a screenshot here for Azure, especially when it would be 95% the same as the one that is now up in the AWS bullet.

Might consider chatting with the previous PR author about whether the AWS screenshot is really needed.

JoeAldinger replied:
I'd be in favor of taking it out. I don't think we normally do screenshots because the GUI changes so often so CI gets to be too tough to keep up with.

JoeAldinger commented on modules/osdk-cco-azure-enabling.adoc:
```diff
@@ -0,0 +1,220 @@
+
+.. Ensure your Operator has RBAC permission to create `CredentialsRequests` objects:
++
+.Example `clusterPermissions` list
```
You've inspired me to use collapsible for long [source,yaml] examples. Nice work.

JoeAldinger reviewed (commented):
A few small suggestions. Clean PR with a lot of sharp prose and great restructuring, nice work.

jianping-shu commented:
@jianzhangbjz PTAL at this doc PR from OLM point of view too, thanks

jianping-shu commented:
Here are my comments:
[CCO-based workflow for OLM-managed Operators with Azure AD Workload Identity](https://70222--ocpdocs-pr.netlify.app/openshift-enterprise/latest/operators/operator_sdk/token_auth/osdk-cco-azure) (fully new)
(1). Precisely this feature is available in 4.14.z. CCO part is in 4.14.8 but I'm not sure about console part.
(2) For below, shall we add Region and CloudTokenPath too? Just confirm
2. Update your Operator project code:
a.Get the client ID, tenant ID, and subscription ID from the environment variables set on the pod by the Subscription object.
(3)The yaml sample in “Example CredentialsRequest object creation in YAML form” is still for AWS instead of Azure
(4)Below was copied from AWS section and shall be reworded for Azure for sample "role ARN -> Azure credential informations"
c. Add the role ARN and web identity token path to the credentials request and apply it during Operator initialization:

[Web console procedure](https://70222--ocpdocs-pr.netlify.app/openshift-enterprise/latest/operators/admin/olm-adding-operators-to-cluster#olm-installing-from-operatorhub-using-web-console_olm-adding-operators-to-a-cluster) (see step 5b)
[CLI procedure](https://70222--ocpdocs-pr.netlify.app/openshift-enterprise/latest/operators/admin/olm-adding-operators-to-cluster#olm-installing-from-operatorhub-using-web-console_olm-adding-operators-to-a-cluster) (see step 5b)
(1)For 5b, shall we add Region and CloudTokenPath too? Just confirm

jianping-shu commented:
@yanpzhan Pls. help review this PR for Console part, thanks

gallettilance reviewed (approved):
/lgtm

jianzhangbjz commented:
/lgtm

adellape commented:
@jianping-shu Thank you! Some replies inline:

> Here are my comments: [CCO-based workflow for OLM-managed Operators with Azure AD Workload Identity](https://70222--ocpdocs-pr.netlify.app/openshift-enterprise/latest/operators/operator_sdk/token_auth/osdk-cco-azure) (fully new) (1). Precisely this feature is available in 4.14.z. CCO part is in 4.14.8 but I'm not sure about console part. 

Changed this reference to "Starting in OpenShift Container Platform 4.14.8 [...]"

> (2) For below, shall we add Region and CloudTokenPath too? Just confirm 2. Update your Operator project code: a.Get the client ID, tenant ID, and subscription ID from the environment variables set on the pod by the Subscription object. 

and

> [Web console procedure](https://70222--ocpdocs-pr.netlify.app/openshift-enterprise/latest/operators/admin/olm-adding-operators-to-cluster#olm-installing-from-operatorhub-using-web-console_olm-adding-operators-to-a-cluster) (see step 5b) [CLI procedure](https://70222--ocpdocs-pr.netlify.app/openshift-enterprise/latest/operators/admin/olm-adding-operators-to-cluster#olm-installing-from-operatorhub-using-web-console_olm-adding-operators-to-a-cluster) (see step 5b) (1)For 5b, shall we add Region and CloudTokenPath too? Just confirm

I didn't see either shown in the video demo from https://github.com/openshift/console/pull/13082. The PR description there also mentions: "we are assuming that token path will be standardized / defaulted to /var/run/secrets/openshift/serviceaccount/token and region is not a required configuration"

> (3)The yaml sample in “Example CredentialsRequest object creation in YAML form” is still for AWS instead of Azure 

I couldn't find an Azure example, so removing the sample for now but can add one in a follow-up PR.

> (4)Below was copied from AWS section and shall be reworded for Azure for sample "role ARN -> Azure credential informations"
c. Add the role ARN and web identity token path to the credentials request and apply it during Operator initialization:

Oops! Fixed.

adellape commented:
/cherrypick enterprise-4.15

adellape commented:
/cherrypick enterprise-4.14
